### PR TITLE
clean up obsolete and red hand related code

### PR DESCRIPTION
**Pull request by JiaoJiao123**
The battery and red hand features have been removed from the curation platform for a while. It is time to clean up all the old features related code such as Obsolete/Unobsolete in Review Mode, Obsolete and Red hand options in gene data import, and Checking Variants by Status in tools page.

### CODE REVIEW

File: web/yo/app/scripts/services/stringutils.js [hunk 1]
```diff
@@ -879,206 +879,174 @@ angular.module('oncokbApp')
...
-                    if (!(excludeObsolete && e.shortSummary_eStatus && e.shortSummary_eStatus.has('obsolete') && e.shortSummary_eStatus.get('obsolete') === 'true')) {
-                        _mutation = combineData(_mutation, e, ['shortSummary', 'oncogenic'], false, excludeComments, onlyReviewedContent);
-                    }
-                    if (!(excludeObsolete && e.oncogenic_eStatus && e.oncogenic_eStatus.has('obsolete') && e.oncogenic_eStatus.get('obsolete') === 'true')) {
-                        _mutation = combineData(_mutation, e, ['description', 'short', 'effect'], excludeObsolete, excludeComments, onlyReviewedContent);
-                        if (e.effect_uuid) {
-                            _mutation.effect_uuid = validUUID(e.effect_uuid);
-                        }
-                        _mutation.effect_review = getReview(e.effect_review);
+                if (onlyReviewedContent && e.name_review.get('added') == true || !onlyReviewedContent && e.name_review.get('removed') == true) return;
```
Comment (zhx828) [inline web/yo/app/scripts/services/stringutils.js]: `&& e.name_review.get('removed') == true` why we have new logic?

File: web/yo/app/scripts/controllers/gene.js [hunk 2]
```diff
@@ -2397,7 +2239,7 @@ angular.module('oncokbApp')
 
             $scope.getAllCitations = function() {
                 var results = [];
-                var geneData = JSON.stringify(stringUtils.getGeneData(this.gene, true, true, true));
+                var geneData = JSON.stringify(stringUtils.getGeneData(this.gene, true, false));
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: Why we change from true to false?

File: web/yo/app/scripts/controllers/gene.js [hunk 3]
```diff
@@ -2464,7 +2306,7 @@ angular.module('oncokbApp')
             $scope.specifyAnnotation = function() {
                 var annotationLocation = {};
                 setAnnotationResult(annotationLocation, fetchResults(FindRegex.result(this.gene.background.text)), 'Gene Background');
-                var mutations = stringUtils.getGeneData(this.gene, false, false, false, false, true).mutations;
+                var mutations = stringUtils.getGeneData(this.gene, true, false).mutations;
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: We only removed `excludeObsolete` and `excludeRedHands`, but why there are three parameters been removed here?

File: web/yo/app/scripts/controllers/gene.js [hunk 4]
```diff
@@ -2656,7 +2498,7 @@ angular.module('oncokbApp')
             };
 
             $scope.generatePDF = function() {
-                jspdf.create(stringUtils.getGeneData(this.gene, true, true, true));
+                jspdf.create(stringUtils.getGeneData(this.gene, true, false));
```
Comment (zhx828) [inline web/yo/app/scripts/controllers/gene.js]: Same here

File: web/yo/app/scripts/services/stringutils.js (hunk 1 above)
Comment (JiaoJiao123) [reply web/yo/app/scripts/services/stringutils.js]: This logic is to apply onlyReviewedContent option to removed sections, which were somehow not included before. For instance, if a tumor is removed but not approved yet, we should still import this tumor if onlyReviewedContent is set to true. In the other way, that tumor will not be imported if onlyReviewedContent is set to false.

File: web/yo/app/scripts/controllers/gene.js (hunk 2 above)
Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/gene.js]: Previously, getGeneData() function took 5 parameters, which means the last parameter onlyReviewedContent is using the default false value.

File: web/yo/app/scripts/controllers/gene.js (hunk 3 above)
Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/gene.js]: Previously, getGeneData() took 5 parameters and it was a mistake to pass 6

File: web/yo/app/scripts/controllers/gene.js (hunk 4 above)
Comment (JiaoJiao123) [reply web/yo/app/scripts/controllers/gene.js]: same with above

**Pull request merged**